compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -136,24 +136,43 @@ LogicalResult distributeLocalMemory(ModuleOp moduleOp) {
     //                     AllocOp and not replace the current AllocOp's use
     //                     within.
     memref::AllocOp newAlloc = memrefToNew[allocOp];
+    ArrayRef<int64_t> newAllocShape = newAlloc.getType().getShape();
     for (Operation *userOp : allocOpUsers) {
       if (auto subviewOp = dyn_cast<memref::SubViewOp>(userOp)) {
         rewriter.replaceAllUsesWith(subviewOp, newAlloc);
         toBeErased.push_back(subviewOp);
       } else if (auto transferReadOp =
                      dyn_cast<vector::TransferReadOp>(userOp)) {
         rewriter.setInsertionPoint(transferReadOp);
-        Value newTransferReadOp = rewriter.create<vector::TransferReadOp>(
+        // For dimensions with size as 1 we set the indices as 0.
```

Comment:
Why is this needed and what are the initial values in `transferReadOp.getIndices()` which are wrong then?

Reply:
Discussed offline and explanation was added